vignettes/gsmExtensions.Rmd

```diff
@@ -0,0 +1,412 @@
+---
+title: "gsm Extensions"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{gsmExtensions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Overview
+
+This vignette describes how to extend `gsm` by creating new "modules", including metrics, reports and shiny apps that can be run using the standard `gsm` pipeline described in these vignettes (`vignette("DataAnalysis")`, `vignette("DataReporting")`). As shown in the `vignette("DataAnalysis")`, the `gsm` data pipeline can be used to capture a monitoring 'snapshot' for a study that includes a variety of "modules" including [metrics](https://github.com/Gilead-BioStats/gsm/tree/dev/inst/workflow/metrics/kri0001.yaml) and [reports](https://github.com/Gilead-BioStats/gsm/blob/dev/inst/workflow/reports/report_kri_site.yaml). Some core modules are included in the `gsm` package, while others can be added as extensions. 
+
+This vignette provide detailed specifications for creating new modules, a description of the directory structure for the yaml workflows that comprise a module pipeline, and links to resources that can be used to configure study-level gsm pipelines that utilize these `gsm` extensions.
+
+# Module Configuration
+
+`gsm` modules will typically be part of an R package and have a YAML configuration file - `{module_name}.yaml` - saved under `\inst\workflow\4_modules`. Module config files include 3 key properties: 
+
+  -  `meta`: (required) Report metadata used for centralized library and fields referenced in `steps` section of workflow.
+  - `spec`: (required) Data table requirement for the report. Typically from `Mapped_*` or `Reporting_*` data. 
+  - `steps`: (required) Functions to produce the reporting output from `Mapped_*` and/or `Reporting_*` data to final output (`html`, `csv` or `shinyApp`).
+   
+Detailed specifications for each of these sections are provided below. 
+
+## Example Module Configuration YAMLs
+
+Here are links to several sample module configuration files:
+
+- [12 Standard gsm KRIs](https://github.com/Gilead-BioStats/gsm/tree/dev/inst/workflow/metrics) (e.g. [Adverse Event KRI Metric](https://github.com/Gilead-BioStats/gsm/tree/dev/inst/workflow/metrics/kri0001.yaml))
+- [Site-](https://github.com/Gilead-BioStats/gsm/blob/dev/inst/workflow/reports/report_kri_site.yaml) and [Country-level KRI Report](https://github.com/Gilead-BioStats/gsm/blob/dev/inst/workflow/reports/report_kri_country.yaml)
+
+## `meta` Specification
+
+The `meta` section of a workflow YAML provides key metadata describing the module. It must include the following fields: 
+
+- `Name`: Name of the reporting output.
+- `Type`: "Report", "Metric" or "App"
+- `ID`: The unique ID for the module. This should be the same as the workflow file name, without extension.
```

Comment:
should we add a "required" tag of some sort for `Type` and `ID` ? As it does trigger those cli_abort()'s in ` MakeWorkflowListI()`

Reply:
The istructions above say that the meta section "must include the following fields", which i think is sufficient